spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/ConverterAwareMappingSpannerEntityWriter.java

```diff
@@ -74,46 +69,41 @@
 
 	@SuppressWarnings("unchecked")
 	private static Map<Class<?>, BiConsumer<ValueBinder<?>, Iterable>> createIterableTypeMapping() {
-		// Java 8 has compile errors when using the builder extension methods
-		// @formatter:off
-		ImmutableMap.Builder<Class<?>, BiConsumer<ValueBinder<?>, Iterable>> builder =
-						new ImmutableMap.Builder<>();
-		// @formatter:on
-
-		builder.put(com.google.cloud.Date.class, ValueBinder::toDateArray);
-		builder.put(Boolean.class, ValueBinder::toBoolArray);
-		builder.put(Long.class, ValueBinder::toInt64Array);
-		builder.put(String.class, ValueBinder::toStringArray);
-		builder.put(Double.class, ValueBinder::toFloat64Array);
-		builder.put(Timestamp.class, ValueBinder::toTimestampArray);
-		builder.put(ByteArray.class, ValueBinder::toBytesArray);
-
-		return builder.build();
+		Map<Class<?>, BiConsumer<ValueBinder<?>, Iterable>> map = new LinkedHashMap<>();
+		map.put(com.google.cloud.Date.class, ValueBinder::toDateArray);
+		map.put(Boolean.class, ValueBinder::toBoolArray);
+		map.put(Long.class, ValueBinder::toInt64Array);
+		map.put(String.class, ValueBinder::toStringArray);
+		map.put(Double.class, ValueBinder::toFloat64Array);
+		map.put(Timestamp.class, ValueBinder::toTimestampArray);
+		map.put(ByteArray.class, ValueBinder::toBytesArray);
+
+		return Collections.unmodifiableMap(map);
 	}
 
 	static {
-		ImmutableMap.Builder<Class<?>, BiFunction<ValueBinder, ?, ?>> builder = new ImmutableMap.Builder<>();
-
-		builder.put(Date.class, (BiFunction<ValueBinder, Date, ?>) ValueBinder::to);
-		builder.put(Boolean.class, (BiFunction<ValueBinder, Boolean, ?>) ValueBinder::to);
-		builder.put(Long.class, (BiFunction<ValueBinder, Long, ?>) ValueBinder::to);
-		builder.put(long.class, (BiFunction<ValueBinder, Long, ?>) ValueBinder::to);
-		builder.put(Double.class, (BiFunction<ValueBinder, Double, ?>) ValueBinder::to);
-		builder.put(double.class, (BiFunction<ValueBinder, Double, ?>) ValueBinder::to);
-		builder.put(String.class, (BiFunction<ValueBinder, String, ?>) ValueBinder::to);
-		builder.put(Timestamp.class,
+		Map<Class<?>, BiFunction<ValueBinder, ?, ?>> map = new LinkedHashMap<>();
```

Comment:
Could this (and the other one in this class) now also use MapBuilder from our util class?

Reply:
I think order matters in `singleItemTypeValueBinderMethodMap` -- `findFirstCompatibleSpannerSingleItemNativeType()`
does an ordered search on it:
````
singleItemTypeValueBinderMethodMap.keySet().stream().filter(testFunc)
				.findFirst();
````

Reply:
I see!